src/google/adk/models/lite_llm.py

```diff
@@ -376,22 +376,137 @@ def _model_response_to_chunk(
 def _model_response_to_generate_content_response(
     response: ModelResponse,
 ) -> LlmResponse:
-  """Converts a litellm response to LlmResponse.
+  """Converts a litellm response to LlmResponse with robust error handling.
+
+  This enhanced version:
+  1. Adds validation for required fields with proper defaults
+  2. Improves JSON parsing to handle both standard JSON and Python-style strings
+  3. Implements comprehensive error handling to prevent crashes
+  4. Maintains compatibility with all model formats
+
+  Implementation note:
+  This function is part of the fix for an infinite loop issue that occurs when using
+  Ollama/Gemma3 models with LiteLLM. These models sometimes return malformed JSON in 
+  function call arguments, which can cause the system to get stuck in a loop.
+  The robust parsing ensures that even with malformed JSON, we can still extract
+  valid arguments and prevent failures.
```

Comment:
While looking at the semantics/style of the rest of the codebase and the fact that you have already provided a separate file documenting these changes, I believe this portion of the function docstring is not necessary.